src/r_draw.c

```diff
@@ -423,50 +426,45 @@ static void R_DrawFuzzColumn_orig(void)
         // why_i_left_doom.html
 
         *dest =
-            fullcolormap[6 * 256
-                         + dest[fuzzoffset[fuzzpos] ? -linesize : linesize]];
+            fullcolormap[6 * 256 + dest[linesize * fuzzoffset[fuzzpos]]];
         dest += linesize; // killough 11/98
 
         ++fuzzpos;
 
-        if (fuzzpos == FUZZTABLE)
-            fuzzpos = 0;
-
+        // Clamp table lookup index.
+        fuzzpos &= (fuzzpos - FUZZTABLE) >> (8 * sizeof(fuzzpos) - 1); // killough 1/99
```

Comment:
I've never understood why Killough wrote this line this way. Obfuscation? Anything wrong with something like this?
```c
fuzzpos = (fuzzpos + 1) % FUZZTABLE;
```

Reply:
I think `%` is more expensive than `>>`, `&`. See "bit twidling" topic, e.g. https://graphics.stanford.edu/~seander/bithacks.html#IntegerAbs

I'm not sure there's much benefit with modern CPU/compilers, though. From my tests, this branch is about ~10% faster at best.